GitUI/CommandsDialogs/RevisionDiffController.cs

```diff
@@ -74,6 +78,96 @@ public sealed class ContextMenuSelectionInfo
 
     public sealed class RevisionDiffController : IRevisionDiffController
     {
+        private readonly Func<IGitModule> _getModule;
+        private readonly IFullPathResolver _fullPathResolver;
+
+        public RevisionDiffController(Func<IGitModule> getModule, IFullPathResolver fullPathResolver)
+        {
+            _getModule = getModule;
+            _fullPathResolver = fullPathResolver;
+        }
+
+        public void SaveFiles(List<FileStatusItem> files, Func<string, string?> userSelection)
+        {
+            if (files is null)
+            {
+                throw new ArgumentNullException(nameof(files));
+            }
+
+            if (files.Count == 0)
+            {
+                return;
+            }
+
+            if (userSelection is null)
+            {
+                throw new ArgumentNullException(nameof(userSelection));
+            }
+
+            if (files.Count > 1)
+            {
+                SaveMultipleFiles(files);
+                return;
+            }
+
+            SaveSingleFile(files[0]);
+            return;
+
+            void SaveMultipleFiles(List<FileStatusItem> selectedFiles)
+            {
+                // Derive the folder from the first selected file.
+                string firstItemFullName = _fullPathResolver.Resolve(selectedFiles.First().Item.Name);
+                string baseSourceDirectory = Path.GetDirectoryName(firstItemFullName).EnsureTrailingPathSeparator();
+
+                string selectedPath = userSelection(baseSourceDirectory);
+                if (selectedPath is null)
+                {
+                    // User has cancelled the selection
+                    return;
+                }
+
+                Uri baseSourceDirectoryUri = new(baseSourceDirectory);
+
+                foreach (FileStatusItem item in selectedFiles)
+                {
+                    string selectedItemFullName = _fullPathResolver.Resolve(item.Item.Name);
+                    string selectedItemSourceDirectory = Path.GetDirectoryName(selectedItemFullName).EnsureTrailingPathSeparator();
+
+                    string targetDirectory;
+                    if (selectedItemSourceDirectory == baseSourceDirectory)
+                    {
+                        targetDirectory = selectedPath;
+                    }
+                    else
+                    {
+                        Uri selectedItemUri = new(selectedItemSourceDirectory);
+                        targetDirectory = Path.Combine(selectedPath, baseSourceDirectoryUri.MakeRelativeUri(selectedItemUri).OriginalString);
+                    }
+
+                    // TODO: check target file exists.
+                    // TODO: allow cancel the whole sequence
```

Comment:
What is the plan?

Reply:
No immediate plans, just noting the future improvement possibilities. This feature is very niche - in 15 years no one really needed it (judging by the lack of requests), and I only needed it few times while sync'ing two codebases.